lib/base-compiler.js

```diff
@@ -210,9 +238,31 @@ Compile.prototype.postProcessAsm = function (result) {
         });
 };
 
+Compile.prototype.processOptoutput = function(hasOptOutput, optPath) {
```

Comment:
Super great job @jaredwy ! Besides what Matt mentioned, (_I'm just nitpicking_) the `o` of `output` should be uppercase. You know.. for consistency :sweat_smile: 

I'll test the code now, but it looks really good already!